src/helper.rs

```diff
@@ -16,3 +17,35 @@ pub struct Node {
     pub udp_port: u16,
     pub node_id: Identifier,
 }
+
+pub mod testing {
+    use super::*;
+
+    pub fn mk_nodes() -> Vec<Node> {
+        // Should these nodes have different IP addresses?
```

Comment:
stylistic thing but I prefer to keep "thoughts" or questions like this out of the code; instead you could make an issue if its worthwhile or in case I'd think you can just keep a list of questions in a separate document locally